File: library/src/main/java/org/meganekkovr/xml/PositionHandler.java
package org.meganekkovr.xml;

import android.content.Context;
import android.support.annotation.NonNull;

import org.meganekkovr.Entity;

/**
 * Define {@code position} attribute.
 */
class PositionHandler implements XmlAttributeParser.XmlAttributeHandler {

    @NonNull
    @Override
    public String attributeName() {
        return "position";
    }

    @Override
    public void parse(@NonNull Entity entity, @NonNull String rawValue, @NonNull Context context) {

        String[] strs = rawValue.split("\\s+", 3);
        if (strs.length == 3) {
            float x = Float.parseFloat(strs[0]);
            float y = Float.parseFloat(strs[1]);
            float z = Float.parseFloat(strs[2]);
            entity.setPosition(x, y, z);
        }
    }
}

File: library/src/main/java/org/meganekkovr/xml/PositionHandler.kt
package org.meganekkovr.xml

import android.content.Context

import org.meganekkovr.Entity

/**
 * Define `position` attribute.
 */
internal class PositionHandler : XmlAttributeParser.XmlAttributeHandler {

    override fun attributeName(): String {
        return "position"
    }

    override fun parse(entity: Entity, rawValue: String, context: Context) {

        val strs = rawValue.split("\\s+".toRegex(), 3)
        if (strs.size == 3) {
            val x = strs[0].toFloatOrNull() ?: return
            val y = strs[1].toFloatOrNull() ?: return
            val z = strs[2].toFloatOrNull() ?: return
            entity.setPosition(x, y, z)
        }
    }
}
